sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2471,7 +2471,7 @@ pi_result piQueueRelease(pi_queue Queue) {
     std::lock_guard<std::mutex> Lock(Queue->PiQueueMutex);
     Queue->RefCount--;
     if (Queue->RefCount == 0)
-      RefCountZero = true;
+      RefCountZero = Queue->OwnZeCommandQueue ? true : false;
```

Comment:
This way you will not execute the entire if below, whereas you only need not run the `zeCommandQueueDestroy` at the end

Reply:
done